test/test_page_without_a_file.rb

```diff
@@ -28,6 +34,44 @@ def render_and_write
                        ))
     end
 
+    should "have non-frozen path and relative_path attributes" do
+      page = PageWithoutAFile.new(@site, @site.source, "foo", "bar.md")
+      assert_equal "foo", page.instance_variable_get(:@dir)
+      assert_equal "bar.md", page.instance_variable_get(:@name)
+      assert_equal "foo/bar.md", page.path
+      assert_equal "foo/bar.md", page.relative_path
+      refute page.relative_path.frozen?
+
+      # reset attributes
+      rehash_page!(page, :dir => nil, :name => nil)
```

Comment:
Instead of doing this, why don't we just create a new PageWithoutAFile instance? I don't think the tests knowing about the internal instance variables is a great idea. Also, These blocks could be separate Tests to help describe what you're testing for.

Reply:
Good question, @parkr.
The reason we can't create a new `PageWithoutAFile` instance is because of the following lines:
https://github.com/jekyll/jekyll/blob/fe898af92fadecbec6b4b2ec69c1cb2d85eb46e2/lib/jekyll/page.rb#L44-L48

`Jekyll::PageWithoutAFile.new(site, site.source, nil, nil)` fails as `Jekyll.sanitized_path` doesn't cast its arguments into strings automatically.

Reply:
`Jekyll.sanitized_path` has been patched via https://github.com/jekyll/jekyll/commit/98b7a4626a86de4e068ffa6bd927f9437a444c28